### PR TITLE
009 add new role

### PR DESCRIPTION
**Pull request by amarauzoma**

### CODE REVIEW

File: src/main/java/org/example/validators/JobRoleValidator.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package org.example.validators;
+
+import org.example.models.JobRoleRequest;
+
+public class JobRoleValidator {
+public boolean validateJobRole(
+        final JobRoleRequest jobRoleRequest) throws Exception {
+       final int maxJobNameLength = 100;
```
Comment (mlynch147) [inline src/main/java/org/example/validators/JobRoleValidator.java]: make these variable STATIC

File: src/main/java/org/example/validators/JobRoleValidator.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+       final int maxLocationLength = 100;
+       final int maxJobDescLength = 100;
+       final int maxJobResponsibilitiesLength = 100;
+       if (jobRoleRequest.getRoleName().length() > maxJobNameLength) {
+           throw new Exception("Role Name greater than 100 characters");
```
Comment (mlynch147) [inline src/main/java/org/example/validators/JobRoleValidator.java]: update all messages to use the variable values

File: src/main/java/org/example/services/JobRoleService.java [hunk 3]
```diff
@@ -52,4 +54,18 @@ public JobRoleDetailedResponse getJobRoleById(final int id)
     public int getTotalRecords() throws SQLException {
         return jobRoleDao.getTotalOpenJobs();
     }
-}
+    public int createJobRole(
+            final JobRoleRequest jobRoleRequest)
+            throws FailedToCreateException, SQLException,
+            DoesNotExistException {
```
Comment (mlynch147) [inline src/main/java/org/example/services/JobRoleService.java]: I cant see anywhere in the code that a DoesNotExistException is thrown in this call stack.  I think you can remove it...

File: src/main/java/org/example/controllers/JobRoleController.java [hunk 4]
```diff
@@ -108,4 +111,28 @@ public Response getJobRoleById(@PathParam("id") final int id) {
...
         }
     }
+    @POST
+    @RolesAllowed({UserRole.ADMIN})
+    @ApiOperation(
+            value = "Add Job Roles",
+            authorizations = @Authorization(value = HttpHeaders.AUTHORIZATION),
+            response = JobRoleRequest.class)
+    public Response createJobRole(final JobRoleRequest jobRoleRequest)
+            throws FailedToCreateException, DoesNotExistException,
```
Comment (mlynch147) [inline src/main/java/org/example/controllers/JobRoleController.java]: I cant see anywhere in the code that a DoesNotExistException is thrown in this call stack.  I think you can remove it...

File: src/main/java/org/example/models/JobRole.java [hunk 5]
```diff
@@ -10,6 +10,8 @@ public class JobRole {
     private String bandName;
     private Date closingDate;
 
+    public JobRole() {
```
Comment (mlynch147) [inline src/main/java/org/example/models/JobRole.java]: Why has this been added ?

Comment (amarauzoma) [reply src/main/java/org/example/models/JobRole.java]: Resolved

File: src/main/java/org/example/controllers/JobRoleController.java (hunk 4 above)
Comment (amarauzoma) [reply src/main/java/org/example/controllers/JobRoleController.java]: Removed

File: src/main/java/org/example/services/JobRoleService.java (hunk 3 above)
Comment (amarauzoma) [reply src/main/java/org/example/services/JobRoleService.java]: Removed

File: src/main/java/org/example/validators/JobRoleValidator.java (hunk 1 above)
Comment (amarauzoma) [reply src/main/java/org/example/validators/JobRoleValidator.java]: resolved

File: src/main/java/org/example/validators/JobRoleValidator.java (hunk 2 above)
Comment (amarauzoma) [reply src/main/java/org/example/validators/JobRoleValidator.java]: resolved

**Review by mlynch147 — APPROVED** [review]

**Review by orlaghmccloskey1 — APPROVED** [review]

**Pull request merged**